govcd/org.go

```diff
@@ -5,29 +5,53 @@
 package govcd
 
 import (
+	"bytes"
+	"encoding/xml"
 	"fmt"
-	"net/url"
-
 	types "github.com/vmware/go-vcloud-director/types/v56"
+	"net/url"
+	"strconv"
+	"strings"
 )
 
 type Org struct {
 	Org *types.Org
 	c   *Client
 }
 
-func NewOrg(c *Client) *Org {
+func NewOrg(client *Client) *Org {
 	return &Org{
 		Org: new(types.Org),
-		c:   c,
+		c:   client,
+	}
+}
+
+// AdminOrg gives an admin representation of an org
+// Users can delete, update orgs with an admin org object
+// AdminOrg users have to get an org representation to use find Catalogs
+// TODO: Discuss whether we want to fill up the org variable as well
+// in the admin org. It will take an extra rest api call to get but it
+// will enable users to be able to make calls to org functions as well.
+// This is probably the closest we can get to pure inheritance.
+type AdminOrg struct {
```

Comment:
Even though there's a comment above the struct, I still find the concept of AdminOrg confusing. It asks for a deeper explanation with some examples.
1.) Does AdminOrg map 1:1 to a particular Org or is it actually the "System" view that you see in the UI? In other words, is it an administrative permission of a particular Org or of the whole system?
2.) Please remove the TODO, as you have an issue #53 filed for it. Better replace those four lines with the wider explanation requested above.
3.) Nit: dots are missing at the end of the comment sentences.

All in all, if you could enhance the comment, so the new contributors get a better sense of what AdminOrg is, it would be very helpful.